src/plugins/azgenerator/CodeModelAzImpl.ts

```diff
@@ -162,6 +177,123 @@ export class CodeModelCliImpl implements CodeModelAz {
             } while (this.SelectNextCommandGroup());
         }
     }
+
+    private setParamAzUniqueNames() {
+        this.paramActionNameReference = new Map<Schema, string>();
+        let nameActionReference: Map<string, ActionParam> = new Map<string, ActionParam>();
+        if (this.SelectFirstCommandGroup()) {
+            do {
```

Comment:
just a refactor suggestion, not specific for this code review, consider to refactor this into enumerable in the future which can make the code cleaner

Reply:
yeah. I've been thinking about it as well. we can use callback